debridge_node/src/modules/chain/scanning/services/SolanaReaderService.ts

```diff
@@ -1,161 +1,127 @@
+    private readonly solanaEventsReaderService: SolanaEventsReaderService,
   ) {
-    this.GET_HISTORICAL_LIMIT = parseInt(readConfiguration(configService, this.logger, 'SOLANA_GET_HISTORICAL_BATCH_SIZE'));
-    this.GET_EVENTS_LIMIT = parseInt(readConfiguration(configService, this.logger, 'SOLANA_GET_EVENTS_BATCH_SIZE'));
-    this.SOLANA_API_WAIT_BATCH_INTERVAL = parseInt(configService.get('SOLANA_API_WAIT_BATCH_INTERVAL') || '1000');
+    this.#solanaGrpcClient = solanaEventsReaderService.getClient();
   }
 
-  /**
-   * Sync transactions
-   * @param {string} chainId
-   */
-  async syncTransactions(chainId: number) {
+  async syncTransactions() {
     const chain = await this.supportedChainRepository.findOne({
       where: {
-        chainId,
+        chainId: solanaChainId,
       },
     });
-    const lastSolanaBlock = await this.solanaApiService.getLastBlock();
-    this.logger.verbose(`lastSolanaBlock = ${lastSolanaBlock}`);
-
-    const latestTransactionInChain = (await this.solanaApiService.getHistoricalData(1))[0]; // get latest transaction
-    let earliestTransactionInSyncSession = undefined;
-    const previousSyncLastTransaction = chain.latestSolanaTransaction;
-
-    if (latestTransactionInChain === previousSyncLastTransaction) {
-      this.logger.log(`Chain ${chainId} is synced`);
-      await this.supportedChainRepository.update(
-        { chainId },
-        {
-          latestBlock: lastSolanaBlock,
-        },
-      );
+    if (!chain) {
+      const message = `Chain ${solanaChainId} is not configured`;
+      this.#logger.error(message);
       return;
     }
-
-    let transactionsHashes: string[] = [];
-    let batchTransactionsHashes: string[];
-    do {
-      batchTransactionsHashes = await this.solanaApiService.getHistoricalData(
-        this.GET_HISTORICAL_LIMIT,
-        earliestTransactionInSyncSession,
-        previousSyncLastTransaction,
-      );
-      if (batchTransactionsHashes.length === 0) {
-        this.logger.log(`Chain ${chainId} is synced`);
+    const submissions = [...this.#submissionsFromSync].filter(s => s.nonce >= chain.latestNonce);
+    submissions.sort((a, b) => a.nonce - b.nonce);
+    this.#submissionsFromSync = [];
+    const size = Math.ceil(submissions.length / this.#PAGE_SIZE);
+    //pagination
+    for (let pageNumber = 0; pageNumber < size; pageNumber++) {
+      const skip = pageNumber * this.#PAGE_SIZE;
+      const end = Math.min((pageNumber + 1) * this.#PAGE_SIZE, submissions.length);
+      const submissionsForProcessing = submissions.slice(skip, end);
+      const lastSubmission = submissionsForProcessing.at(-1);
+      const processingResult = await this.chainProcessingService.process(submissionsForProcessing, solanaChainId, lastSubmission.nonce);
+      if (processingResult === ProcessNewTransferResultStatusEnum.ERROR) {
+        this.#submissionsFromSync = [];
+        this.createSubscription();
```

Comment:
Can you describe this logic. It's not clear for me. Why do you create subscribe if sync process return error?